client/txn.go

```diff
@@ -76,6 +76,7 @@ type Txn struct {
 	// is (still) unset.
 	haveTxnWrite bool
 	haveEndTxn   bool // True if there was an explicit EndTransaction
+	execing      bool // True if the txn is run in the exec() loop
```

Comment:
as the immediate hack i think i'd rather have a `CommitNoRollback` for that test, but maybe it's not that important.